elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/TableContext.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2021, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.datastores.aggregation.metadata;
+
+import static com.yahoo.elide.datastores.aggregation.metadata.ColumnContext.ARGS_KEY;
+import static com.yahoo.elide.datastores.aggregation.metadata.ColumnContext.TBL_PREFIX;
+
+import com.yahoo.elide.core.request.Argument;
+import com.yahoo.elide.datastores.aggregation.query.Query;
+import com.yahoo.elide.datastores.aggregation.query.Queryable;
+import com.yahoo.elide.datastores.aggregation.queryengines.sql.metadata.SQLTable;
+
+import com.github.jknack.handlebars.EscapingStrategy;
+import com.github.jknack.handlebars.Formatter;
+import com.github.jknack.handlebars.Handlebars;
+import com.github.jknack.handlebars.HandlebarsException;
+import com.github.jknack.handlebars.Template;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.ToString;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Context for resolving table arguments in provided expression.
+ */
+@Getter
+@ToString
+@Builder
+public class TableContext extends HashMap<String, Object> {
+
+    private final Queryable queryable;
+
+    private final Handlebars handlebars = new Handlebars()
+                    .with(EscapingStrategy.NOOP)
+                    .with((Formatter) (value, next) -> {
+                        if (value instanceof com.yahoo.elide.core.request.Argument) {
+                            return ((com.yahoo.elide.core.request.Argument) value).getValue();
+                        }
+                        return next.format(value);
+                    });
+
+    public Object get(Object key) {
+
+        if (key.equals(TBL_PREFIX)) {
+            return this;
```

Comment:
This would allow things like $$table.$$table.foo.  Minor but technically not correct.